packages/render-fidelity-tools/src/components/renderers/filament-viewer.ts

```diff
@@ -71,12 +72,17 @@ export class FilamentViewer extends LitElement {
   private[$canvas]: HTMLCanvasElement|null = null;
   private[$boundingBox]: Aabb = {min: [0, 0, 0], max: [0, 0, 0]};
 
+  private[$initPromise]: Promise<void>|null = null;
+
   constructor() {
     super();
 
-    init([], () => {
-      this[$initialize]();
-    });
+    this[$initPromise] = new Promise((resolve) => {
```

Comment:
Not sure if i used promise correctly. Also, is it okay if a promise doesn't call reject()?

Reply:
This is probably fine; I think the problem is just that this'll hang if the $initialize function hangs. The robust way is to wrap it in a setTimeout that will reject if it takes too long, but hopefully we don't require that kind of hardening here.